controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -722,6 +715,32 @@ private boolean isSubscriberToNonSubscriberTransition(final ReaderGroupConfigRec
            ReaderGroupState.DELETING, currentState, context, executor));
     }
 
+    /**
+     * Delete Scope Recursively.
+     *
+     * @param scope           scope to be deleted.
+     * @param requestId       request id.
+     * @return deletion status.
+     */
+    public CompletableFuture<DeleteScopeRecursiveStatus.Status> deleteScopeRecursive(final String scope,
+                                                                              final long requestId) {
+        final OperationContext context = streamMetadataStore.createScopeContext(scope, requestId);
+        return RetryHelper.withRetriesAsync(() ->
+                streamMetadataStore.checkScopeExists(scope, context, executor)
+                        .thenCompose(exists -> {
+                            if(!exists){
+                                return CompletableFuture.completedFuture(DeleteScopeRecursiveStatus.Status.SCOPE_NOT_FOUND);
+                            } else {
+                                DeleteScopeEvent deleteEvent = new DeleteScopeEvent(scope, requestId);
+                                return eventHelper.addIndexAndSubmitTask(deleteEvent,
+                                                            () -> streamMetadataStore.setScopeState(scope, State.SEALING, context, executor))
```

Comment:
The scope entity does not have any concept of STATE and I don't think we should be adding it just for this fix.